domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+   *   - id:29389, machine: example_org, domain: example.org  (default)
+   *   - id:91736, machine: example_net, domain: example.net
+   *
+   * @param $domain_id
+   *   The numeric id, machine name, or hostname of the domain to delete. The
+   *   value "all" is taken to mean delete all except the default domain.
+   * @param array $options
+   *    An associative array of options whose values come from cli, aliases,
+   *    config, etc.
+   *
+   * @usage drush domain:delete example.com
+   *   Delete the domain example.com, assigning its content and users to
+   *   the default domain, example.org.
+   *
+   * @usage drush domain:delete --content-as=ignore example.com
+   *   Delete the domain example.com, leaving its content untouched but
+   *   assigning its users to the default domain.
+   *
+   * @usage drush domain:delete --content-as=example_net --users-as=example_net
+   *   example.com Delete the domain example.com, assigning its content and
+   *   users to the example.net domain.
+   *
+   * @usage drush domain:delete --dryrun 19476
+   *   Show the effects of delete the domain example.com and assigning its
+   *   content and users to the default domain, example.org, but not doing so.
+   *
+   * @usage drush domain:delete --chatty example_net
+   *   Verbosely Delete the domain example.net and assign its content and users
+   *   to the default domain, example.org.
+   *
+   * @usage drush domain-delete --chatty all
+   *   Verbosely Delete the domains example.com and example.net and assign
+   *   their content and users to the default domain, example.org.
+   *
+   * @option chatty
+   *   Document each step as it is performed.
+   * @option dryrun
+   *   Do not do anything, but explain what would be done. Implies --chatty.
+   * @option content-as
+   *   Values "prompt", "ignore", "default", <name>, Reassign content
+   *   associated with the the domain being deleted to the default domain, to
+   *   a specified domain, or leave the content alone (&hence inaccessible in
+   *   the normal way). The default value is 'prompt': ask which domain to use.
+   * @option users-as
+   *   Values "prompt", "ignore", "default", <name>, Reassign user accounts
+   *   associated with the the domain being deleted to the default domain,
+   *   to the domain whose machine name is <name>, or leave the user accounts
+   *   alone (&hence inaccessible in the normal way). The default value is
+   *   'prompt': ask which domain to use.
+   *
+   * @command domain:delete
+   * @aliases domain-delete
+   * 
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function delete($domain_id, $options = ['content-as' => null, 'users-as' => null, 'dryrun' => null, 'chatty' => null ]) {
+    $policy_content = 'prompt';
+    $policy_users = 'prompt';
+
+    $this->is_dry_run = (bool)$options['dryrun'];
+
+    //region Get current domain list and perform sanity checks.
```

Comment:
We don't use // region annotations.

I also prefer not to use the term `sanity check` and prefer `validations` or similar.